src/server.c

```diff
@@ -680,6 +680,50 @@ int clientsCronResizeQueryBuffer(client *c) {
     return 0;
 }
 
+/* The client output buffer can be adjusted to better fit the memory requierments.
+ *
+ * the logic is:
+ * in case the last observed peak size of the buffer equals the buffer size - we double the size
+ * in case the last observed peak size of the buffer is less than half the buffer size - we shrink by half.
+ * in any case we will reset the peak value each cron run.
+ * The function always returns 0 as it never terminates the client. */
+int clientsCronResizeOutputBuffer(client *c) {
+
+    void *new_buf = NULL;
+    const size_t BUFFER_TARGET_SHRINK_SIZE = c->buf_usable_size/2;
+    const size_t BUFFER_TARGET_EXPEND_SIZE = c->buf_usable_size*2;
```

Comment:
I don't like that these are upper case. They are variables and should be lower case.